probe-rs/src/memory/adi_v5_memory_interface.rs

```diff
@@ -148,25 +150,35 @@ impl ADIMemoryInterface {
     pub fn read_block32<AP>(
         &self,
         debug_port: &mut AP,
-        address: u32,
+        start_address: u32,
         data: &mut [u32],
     ) -> Result<(), AccessPortError>
     where
         AP: APAccess<MemoryAP, CSW> + APAccess<MemoryAP, TAR> + APAccess<MemoryAP, DRW>,
     {
-        if (address % 4) != 0 {
+        if (start_address % 4) != 0 {
             return Err(AccessPortError::MemoryNotAligned);
         }
 
         // Second we read in 32 bit reads until we have less than 32 bits left to read.
         let csw = self.build_csw_register(DataSize::U32);
         self.write_register_ap(debug_port, csw)?;
 
+        let mut address = start_address;
         let tar = TAR { address };
         self.write_register_ap(debug_port, tar)?;
 
         for data in data.iter_mut() {
             *data = self.read_register_ap(debug_port, DRW::default())?.data;
+
+            // the autoincrement is limited to the 10 lowest bits so we need to write the address
+            // every time it overflows
+            address += 4;
+            if address & 0x3FF == 0 {
+                info!("Writing TAR: {:08x}", address);
```

Comment:
Should I remove the log messages?

Reply:
Maybe make it a `debug` or `trace` instead? That's less noisy

Reply:
I used info because debug was sooo noisy :wink: but yes. Fixed.